multimap/src/main/java/org/infinispan/multimap/impl/ScoredValue.java

```diff
@@ -0,0 +1,77 @@
+package org.infinispan.multimap.impl;
+
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.multimap.impl.internal.MultimapObjectWrapper;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+
+import java.util.Objects;
+
+@ProtoTypeId(ProtoStreamTypeIds.MULTIMAP_SCORED_VALUE)
+public class ScoredValue<V> implements Comparable<ScoredValue<V>> {
+   private static final Object NO_VALUE = new Object();
+   private final Double score;
+   private final MultimapObjectWrapper<V> value;
+
+   public static <V> ScoredValue<V> of(double score, V value) {
+      return new ScoredValue<>(score, new MultimapObjectWrapper<>(value));
+   }
+
+   public static <V> ScoredValue<V> of(double score) {
+      return new ScoredValue<>(score, new MultimapObjectWrapper(NO_VALUE));
```

Comment:
just curious why these are needed?

Reply:
for the sorting issue and finding range elements